src/vueWrapper.ts

```diff
@@ -1,35 +1,37 @@
-import { ComponentPublicInstance, nextTick, App } from 'vue'
+import {
+  nextTick,
+  App,
+  ComponentCustomProperties,
+  ComponentPublicInstance
+} from 'vue'
 import { ShapeFlags } from '@vue/shared'
 // @ts-ignore todo - No DefinitelyTyped package exists for this
 import pretty from 'pretty'
 
 import { config } from './config'
 import domEvents from './constants/dom-events'
 import { DOMWrapper } from './domWrapper'
-import {
-  FindAllComponentsSelector,
-  FindComponentSelector,
-  VueElement
-} from './types'
-import { createWrapperError } from './errorWrapper'
-import { find, matches } from './utils/find'
+import { VueElement } from './types'
 import { mergeDeep } from './utils'
 import { emitted, recordEvent } from './emit'
 import BaseWrapper from './baseWrapper'
-import WrapperLike from './interfaces/wrapperLike'
 
-export class VueWrapper<T extends ComponentPublicInstance>
-  extends BaseWrapper<T['$el']>
-  implements WrapperLike
-{
+export class VueWrapper<
+  T extends Omit<
+    ComponentPublicInstance,
+    '$emit' | keyof ComponentCustomProperties
+  > & {
+    $emit: (event: any, ...args: any[]) => void
```

Comment:
Should this be `event: string`?

Reply:
Unfortunately no. `ComponentPublicInstance<>` (with all default parameters) infers $emit exactly as `(event: string, ...args: any[]) => void`. When you pass here a component with defined emits (for example `hi`) $emit will be `(event: "hi") => void'` which is **not** a subtype of `(event: string, ...args: any[]) => void` (event argument is too narrow). That's why we need to cheat with `any`

I've tried to add `event: EmitType` and `EmitType extends string = string` as parameter to VueWrapper generic to express this limitiation , but it seems typescript still infers wrong type